src/ml/neural_net/tf_compute_context.cpp

```diff
@@ -269,5 +268,56 @@ tf_model_backend::~tf_model_backend() {
   call_pybind_function([&]() { model_ = pybind11::object(); });
 }
 
+tf_image_augmenter::tf_image_augmenter(const options& opts)
+    : processed_image_augmenter(opts) {}
+
+image_augmenter::result tf_image_augmenter::prepare_images(
+    std::vector<labeled_image> source_batch) {
+  return processed_image_augmenter::prepare_images(source_batch);
+}
+
+image_augmenter::intermediate_result
+tf_image_augmenter::prepare_augmented_images(
+    image_augmenter::intermediate_labeled_image data_to_augment) {
+  // const size_t n = opts_.batch_size;
+  image_augmenter::intermediate_result image_annotations;
+
+  call_pybind_function([&]() {
+    // Pass the required data to tensorflow
+    pybind11::module tf_aug = pybind11::module::import(
+        "turicreate.toolkits.object_detector._tf_image_augmenter");
+
+    // Get augmented images and annotations from tensorflow
+    pybind11::object tf_augmentation = tf_aug.attr("DataAugmenter")(
+        data_to_augment.images, data_to_augment.annotations,
+        data_to_augment.predictions);
+    pybind11::object augmented_img =
+        tf_augmentation.attr("get_augmented_images")();
+    pybind11::buffer aug_images = augmented_img.cast<pybind11::buffer>();
+    pybind11::buffer_info buf_img = aug_images.request();
+    image_annotations.images = turi::neural_net::shared_float_array::copy(
+        static_cast<float*>(buf_img.ptr),
+        std::vector<size_t>(buf_img.shape.begin(), buf_img.shape.end()));
+    pybind11::object augmented_ann =
+        tf_augmentation.attr("get_augmented_annotations")();
+    std::vector<pybind11::buffer> aug_annotations =
+        augmented_ann.cast<std::vector<pybind11::buffer>>();
+    std::vector<turi::neural_net::shared_float_array> annotations_per_batch;
+    for (size_t i = 0; i < aug_annotations.size(); i++) {
+      pybind11::buffer_info buf_ann = aug_annotations[i].request();
+      turi::neural_net::shared_float_array annotate =
+          turi::neural_net::shared_float_array::copy(
+              static_cast<float*>(buf_ann.ptr),
+              std::vector<size_t>(buf_ann.shape.begin(), buf_ann.shape.end()));
+      annotations_per_batch.push_back(annotate);
+    }
+    image_annotations.annotations = annotations_per_batch;
+  });
+
+  return image_annotations;
+}
+
+tf_image_augmenter::~tf_image_augmenter() {}
```

Comment:
Why add a destructor if it's empty?

Reply:
It is not needed. I removed it now and added the change Nick suggested in the header.